src/main/java/com/activestate/cloudfoundryjenkins/CloudFoundryPushPublisher.java

```diff
@@ -196,156 +217,172 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
-                }
-                listener.getLogger().println("Cloud Foundry push successful.");
-                return true;
-            } else {
-                listener.getLogger().println("ERROR: The application failed to start after " + TIMEOUT + " seconds.");
-                listener.getLogger().println("Cloud Foundry push failed.");
-                return false;
-            }
-        } catch (CloudFoundryException e) {
-            listener.getLogger().println("ERROR: Unknown CloudFoundryException: " + e.getMessage());
-            listener.getLogger().println("ERROR: Cloud Foundry error code: " + e.getCloudFoundryErrorCode());
-            if (e.getDescription() != null) {
-                listener.getLogger().println("ERROR: " + e.getDescription());
-            }
-            e.printStackTrace(listener.getLogger());
-            return false;
-        } catch (FileNotFoundException e) {
-            listener.getLogger().println("ERROR: Could not find file: " + e.getMessage());
-            return false;
-        } catch (ZipException e) {
-            listener.getLogger().println("ERROR: ZipException: " + e.getMessage());
-            return false;
-        }
-    }
+ BuildListener listener, URL targetUrl) throws IOException,
+			InterruptedException {
+		try {
+			String OrigAppURI = "https://" + deploymentInfo.getHostName() + "." + deploymentInfo.getDomain();
+			System.out.println("The Orig Route  "+OrigAppURI);
+			listener.getLogger().println("Pushing " + deploymentInfo.getAppName() + " app to " + target);
+
+			// This is where we would create services, if we decide to add that feature.
+			// List<CloudService> cloudServices = deploymentInfo.getServices();
+			// client.createService();
+			String greenDeployAppName = handleExistingAppAndUpdateDeploymentInfo(client, listener, deploymentInfo);
+			String appURI = "https://" + deploymentInfo.getHostName() + "." + deploymentInfo.getDomain();
+			
+			addToAppURIs(appURI);
+			addToAppURIs(OrigAppURI);
+			if (deploymentInfo.isNewAppToBeCreated()) {
+				// Create app if it doesn't already exist, or if resetIfExists parameter is true
+				createApplication(client, listener, deploymentInfo, appURI);
+			}
+			String appName = deploymentInfo.getAppName();
+			// Unbind all routes if no-route parameter is set
+			if (deploymentInfo.isNoRoute()) {
+				client.updateApplicationUris(appName, new ArrayList<String>());
+			}
+
+			// Add environment variables
+			if (!deploymentInfo.getEnvVars().isEmpty()) {
+				client.updateApplicationEnv(appName, deploymentInfo.getEnvVars());
+			}
+			
+			// Change number of instances
+			if (deploymentInfo.getInstances() > 1) {
+				client.updateApplicationInstances(appName, deploymentInfo.getInstances());
+			}
+			//Assigning the Original Route To Green Deployment
+			//TODO: Need a better way to do same as during 
+			addRouteToApplication(client, greenDeployAppName, OrigAppURI);
```

Comment:
this should not be done here. the route remap should be done once the push is fully done.